### PR TITLE
Fix formatting around the VB dictionary access operator

### LINKED ISSUE #8258: VB: Dictionary access operator formats incorrectly
```VB
Class C
    Public Shared Sub AutoFormatSample()

        ' Automatic Code Formatting in VB.NET
        ' Problem with ! dictionary access operator

        Dim dict As New Dictionary(Of String, String)
        dict.Add("Apple", "Green")
        dict.Add("Orange", "Orange")
        dict.Add("Banana", "Yellow")

        Dim x As String = (New Dictionary(Of String, String)(dict)) !Banana 'added space in front of "!"

        With dict

            !Apple = "Red"
            Dim multiColors = !Apple &!Orange &!Banana 'missing space between "&" and "!"
            If!Banana = "Yellow" Then 'missing space between "If" and "!"
                !Banana = "Green"
            End If

        End With

    End Sub
End Class
```

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes #8258

Tagging @dotnet/roslyn-ide

### CODE REVIEW

**Comment by jasonmalinowski** [comment]
:+1:

**Comment by DustinCampbell** [comment]
Looks like I broke dictionary operators next to conditional access expressions. Will fix.

File: src/Workspaces/VisualBasicTest/Formatting/FormattingTests.vb [hunk 1]
```diff
@@ -4356,5 +4356,66 @@ End Class</Code>
 
             Await AssertFormatLf2CrLfAsync(code.Value, code.Value)
         End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.Formatting)>
+        <WorkItem(8258, "https://github.com/dotnet/roslyn/issues/8258")>
+        Public Async Function FormatDictionaryOperatorProperly() As Task
```
Comment (rchande) [inline src/Workspaces/VisualBasicTest/Formatting/FormattingTests.vb]: Can you put this in a CDATA to get rid of the escaped lt/gt brackets?

**Comment by rchande** [comment]
:+1:

File: src/Workspaces/VisualBasicTest/Formatting/FormattingTests.vb (hunk 1 above)
Comment (DustinCampbell) [inline src/Workspaces/VisualBasicTest/Formatting/FormattingTests.vb]: Are you OK if I just put it in a multiline string? So much nicer!

**Pull request merged**